src/core/lib/surface/server.cc

```diff
@@ -1186,14 +1186,9 @@ void grpc_server_setup_transport(
     chand->channelz_socket_uuid = 0;
   }
 
-  size_t cq_idx;
-  for (cq_idx = 0; cq_idx < s->cq_count; cq_idx++) {
-    if (grpc_cq_pollset(s->cqs[cq_idx]) == accepting_pollset) break;
-  }
-  if (cq_idx == s->cq_count) {
-    /* completion queue not found: pick a random one to publish new calls to */
-    cq_idx = static_cast<size_t>(rand()) % s->cq_count;
-  }
+  const size_t cq_idx = gpr_murmur_hash3(&transport, sizeof(grpc_transport*),
+                                         grpc_core::g_hash_seed) &
+                        (s->cq_count - 1);
```

Comment:
I need to review this change carefully because of the impact on sync/async APIs as well.

Reply:
One thought would be to have a flag that limits this to CB mode only. We can discuss it tomorrow.

Reply:
And by tomorrow I guess I mean next Tuesday.